src/rtutils.c

```diff
@@ -846,7 +846,14 @@ static size_t jl_static_show_x_(JL_STREAM *out, jl_value_t *v, jl_datatype_t *vt
             n += jl_printf(out, "Symbol(\"");
         else
             n += jl_printf(out, ":");
-        n += jl_printf(out, "%s", sn);
+        char buf[64];
+        size_t i = 0, ninc = 0, sz = strlen(sn);
+        while (i < sz) {
+            ninc = u8_escape(buf, sizeof(buf), sn, &i, sz, 1, 0) - 1;
+            n += ninc;
+            jl_uv_puts(out, buf, ninc);
+        }
+        n += sz;
```

Comment:
Isn't the `n += ninc` inside the loop sufficient to count the amount of output?

Reply:
Also would be good to use this code for showing strings as well.

Reply:
seems like a good thing to put in a helper function